### PR TITLE
Update header to 1.2.176

### PR DESCRIPTION
**Pull request by mikes-lunarg**
- Update known-good
- Generate source
- Add validate_bool32_array helper to stateless validation for use in
  vkCmdSetColorWriteEnableEXT
- Changes to how layer_chassis_dispatch_generator.py determines which pNext chains contain handles

There is one VUID inconsistency which will be addressed in a follow-on commit: `VUID-VkRectLayerKHR-offset-01261` became `VUID-VkRectLayerKHR-offset-04864`

### CODE REVIEW

File: layers/stateless_validation.h [hunk 1]
```diff
@@ -644,6 +644,42 @@ class StatelessValidation : public ValidationObject {
...
+     * @param array Array of VkBool32 values to validate.
+     * @param countRequired The 'count' parameter may not be 0 when true.
+     * @param arrayRequired The 'array' parameter may not be NULL when true.
+     * @return Boolean value indicating that the call should be skipped.
+     */
+    bool validate_bool32_array(const char *apiName, const ParameterName &countName, const ParameterName &arrayName, uint32_t count,
+                               const VkBool32 *array, bool countRequired, bool arrayRequired) const {
+        bool skip_call = false;
+
+        if ((count == 0) || (array == NULL)) {
```
Comment (ncesario-lunarg) [inline layers/stateless_validation.h:665]: `NULL` -> `nullptr`?

**Review by ncesario-lunarg — APPROVED** [review]
LGTM!

**Review by mark-lunarg — APPROVED** [review]

File: layers/stateless_validation.h (hunk 1 above)
Comment (mikes-lunarg) [reply layers/stateless_validation.h:665]: Good point, that's what I get for Ctrl-C Ctrl-V

Comment (mikes-lunarg) [reply layers/stateless_validation.h]: Pushed 2123dec718d532fefe72849a1e7575d478e49ae5 to replace `NULL` comparisons with `nullptr`

**Comment by mikes-lunarg** [comment]
Latest commit passed internal CI

**Pull request merged**